src/controllers/keyboard/keyboardeventfilter.cpp

```diff
@@ -156,15 +180,139 @@ QKeySequence KeyboardEventFilter::getKeySeq(QKeyEvent* e) {
 #endif
 }
 
-void KeyboardEventFilter::setKeyboardConfig(ConfigObject<ConfigValueKbd>* pKbdConfigObject) {
+void KeyboardEventFilter::setEnabled(bool enabled) {
+    m_enabled = enabled;
+    m_pConfig->set(ConfigKey("[Keyboard]", "Enabled"), ConfigValue(enabled));
+    emit shortcutsEnabled(enabled);
+}
+
+void KeyboardEventFilter::registerShortcutWidget(WBaseWidget* pWidget) {
+    m_widgets.append(pWidget);
+
+    // Tell widgets to reconstruct tooltips when option is toggled.
+    // WBaseWidget is not a QObject, need to use a lambda
+    connect(this,
+            &KeyboardEventFilter::shortcutsEnabled,
+            this,
+            [pWidget](bool enabled) {
+                pWidget->toggleKeyboardShortcutHints(enabled);
+            });
+}
+
+void KeyboardEventFilter::updateWidgetShortcuts() {
+    for (auto pWidget : m_widgets) {
```

Comment:
Looks like there is a clang-tidy here
```suggestion
    for (auto* pWidget: m_widgets) {
```